test/functional/feature_startupnotify.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -startupnotify."""
+
+import platform
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+
+INVALID_COMMAND = "abc"
+
+
+class StartupNotifyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.disable_syscall_sandbox = True
+        if platform.system() == "Windows":
+            self.extra_args = [["-startupnotify=dir"]]
+        else:
+            self.extra_args = [["-startupnotify=ls"]]
+
+    def run_test(self):
+        self.log.info("Test the node starts with a valid command on -startupnotify")
+        self.stop_node(0)
+        self.log.info("Test the node starts with an invalid command on -startupnotify")
+        self.start_node(0, extra_args=[f"-startupnotify={INVALID_COMMAND}"])
+
+        stderrs = [f"sh: {INVALID_COMMAND}: command not found", f"sh: 1: {INVALID_COMMAND}: not found",
+        f"AssertionError: Unexpected stderr '{INVALID_COMMAND}' is not recognized as an internal or external command,{os.linesep}operable program or batch file."]
```

Comment:
In e9ebe3b, shouldn't it be this?

```suggestion
        f"'{INVALID_COMMAND}' is not recognized as an internal or external command,{os.linesep}operable program or batch file."]
```

Reply:
Not sure if the right is:  
`f"'{INVALID_COMMAND}' is not recognized as an internal or external command,{os.linesep}operable program or batch file."]` or `f"Unexpected stderr '{INVALID_COMMAND}' is not recognized as an internal or external command,{os.linesep}operable program or batch file."`

Going to test it

Reply:
Both of them didn't work.

Reply:
It should be the first option because "AssertionError: Unexpected stderr" comes from [L351](https://github.com/bitcoin/bitcoin/blob/7f462b1a95a48762186783fadf01ae8f3c6df56f/test/functional/test_framework/test_node.py#L351). There shouldn't be a space before `{os.linesep}` in  [L31](https://github.com/bitcoin/bitcoin/blob/7f462b1a95a48762186783fadf01ae8f3c6df56f/test/functional/feature_startupnotify.py#L31)?

Reply:
My bad, I hadn't notice this one on L351. Going to test again, thanks!